components/external_storage/src/s3.rs

```diff
@@ -255,9 +251,9 @@ mod tests {
             )
             .unwrap();
 
-        let mut reader = storage.read("huge_file").unwrap();
+        let mut reader = storage.read("huge_file");
         let mut buf = Vec::new();
-        futures::executor::block_on(reader.read_to_end(&mut buf)).unwrap();
+        block_on_external_io(reader.read_to_end(&mut buf)).unwrap();
```

Comment:
Curious about why this test does not fail before?

Reply:
Because this is in a test case relying on an external S3 server and thus explicitly disabled.